addons/website_crm/controllers/main.py

```diff
@@ -50,6 +50,9 @@ def _handle_website_form(self, model_name, **kwargs):
     def insert_record(self, request, model, values, custom, meta=None):
         is_lead_model = model.model == 'crm.lead'
         if is_lead_model:
+            logged_partner = request.env['website.visitor']._get_visitor_from_request().partner_id
```

Comment:
Technically you may receive None as a result (should not happen in real use case as it happens only if request is not defined). I would maybe say
```
visitor = request.env['website.visitor']._get_visitor_from_request()
if visitor and visitor.partner_id:
    values['partner_id'] = visitor.partner_id
```

Reply:
Good point, I wondered if in this kind of situation I had to "play safe" ? Should I make the change ? Thx :)

Reply:
Well here being safe is not really costly, so yes better be sure.

Reply:
Ok, well noted @tde-banana-odoo  ! I will include it and ping you when all is done